src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -2133,11 +2133,16 @@ public void Dispose()
             }
         }
 
-        public async Task<(int Index, ServerNode Node)> GetRequestedNode(string nodeTag)
+        public async Task<(int Index, ServerNode Node)> GetRequestedNode(string nodeTag, bool checkIfAvailable = false)
         {
             await EnsureNodeSelector().ConfigureAwait(false);
+            (var index, var node) = _nodeSelector.GetRequestedNode(nodeTag);
+            if (checkIfAvailable && !_nodeSelector.NodeIsAvailable(index))
```

Comment:
we use the convention of `_nodeSelector.NodeIsAvailable(index) == false` instead of using `!`